### PR TITLE
DCS-47: Content in Check Answers page now conforms to capitalisation …

### PR DESCRIPTION
**Pull request by GurnankCheema**
…rules

http://localhost:3000/tasklist/-1

Content designer requested capitalisation of values in Check Answers. Previously they were a mix of upper and lower cases.

All none-person name fields begin with a capital, person name fields have a capital first letter on every word.

Changed the functionality in check-answers.html so that different Nunjucks filters can be used depending on the data being presented. The knock on of having individual lines of code for each field was that one set of looping (see {% for key, value in data %}) in pagesMacro.njk was not required.
 
Updated Cypress tests to use 'data-qa' rather than css selectors

### CODE REVIEW

File: server/views/pages/pagesMacros.njk [hunk 1]
```diff
@@ -1,32 +1,31 @@
...
+{% macro tableRow(dataItem) %}
+ <dl class="govuk-summary-list govuk-!-margin-0">  
+        <div class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key"> {{dataItem.label}} </dt>
+            <dd class="govuk-summary-list__value" data-qa="{{dataItem['data-qa']}}">
+              {% if dataItem.dataValue.length === 0 %}
+                &ndash;
+              {% elif dataItem.type === 'bagAndTag' %}
+                {%for item in dataItem.dataValue %}
+                {{item.tag}}<br/>{{item.description | dump}} <br/><br/>
```
Comment (andrewrlee) [inline server/views/pages/pagesMacros.njk]: I'd rename `bagAndTag` to `itemPerLine` or something more generic
You also have a dump in here still

Comment (GurnankCheema) [reply server/views/pages/pagesMacros.njk]: oops!
I'll change that now

Comment (GurnankCheema) [reply server/views/pages/pagesMacros.njk]: Changes made

File: server/routes/checkAnswers.js [hunk 2]
```diff
@@ -118,61 +112,60 @@ const typeOfHandcuffsUsed = handcuffsType => {
...
 
 const createRelocationObj = (relocationAndInjuries = {}) => {
   return {
-    'Prisoner relocated to': relocationAndInjuries.prisonerRelocation,
-    'Relocation compliancy': relocationAndInjuries.relocationCompliancy,
-    'HealthCare staff present': whenPresent(relocationAndInjuries.healthcareInvolved, value =>
-      value === 'Yes' ? `${toTitleCase(relocationAndInjuries.healthcarePractionerName)}` : 'No'
+    prisonerRelocation: relocationAndInjuries.prisonerRelocation,
+    prisonerCompliancy: relocationAndInjuries.relocationCompliancy,
+    halthcareStaffPresent: whenPresent(relocationAndInjuries.healthcareInvolved, value =>
```
Comment (petergphillips) [inline server/routes/checkAnswers.js]: tpyo

Comment (petergphillips) [reply server/routes/checkAnswers.js]: should have caused the cyprus test to fail though :thinking:

File: integration-tests/pages/checkAnswersPage.js [hunk 3]
```diff
@@ -5,40 +5,41 @@ const clickSubmit = () => cy.get('[data-submit]').click()
...
-      cy.get(':nth-child(5) > :nth-child(7) > .govuk-summary-list__value').contains('ratchet')
-      cy.get(':nth-child(7) > :nth-child(1) > .govuk-summary-list__value').contains('segregation unit')
-      cy.get(':nth-child(7) > :nth-child(2) > .govuk-summary-list__value').contains('compliant')
-      cy.get(':nth-child(7) > :nth-child(3) > .govuk-summary-list__value').contains('Dr Smith')
-      cy.get(':nth-child(7) > :nth-child(4) > .govuk-summary-list__value').contains('Dr Taylor')
-      cy.get(':nth-child(7) > :nth-child(5) > .govuk-summary-list__value').contains('Yes')
-      cy.get(':nth-child(7) > :nth-child(6) > .govuk-summary-list__value').contains('Eddie Thomas, Jayne Eyre')
-      cy.get(':nth-child(7) > :nth-child(6) > .govuk-summary-list__value').contains('Eddie Thomas, Jayne Eyre')
-      cy.get(':nth-child(9) > :nth-child(1) > .govuk-summary-list__value')
+      cy.get('[data-qa="offenderName"]').contains('Norman Smith')
```
Comment (petergphillips) [inline integration-tests/pages/checkAnswersPage.js]: much better :grin:

File: server/routes/checkAnswers.js (hunk 2 above)
Comment (GurnankCheema) [reply server/routes/checkAnswers.js]: Change made and pushed up

File: server/views/pages/check-answers.html [hunk 4]
```diff
@@ -65,7 +65,7 @@ <h1 class="govuk-heading-xl">Check your answers before sending the report</h1>
           }}
       {{pagesMacros.tableRow({label: 'Prisoner relocated to',             'data-qa': 'prisonerRelocation',      dataValue: data.relocationAndInjuries.prisonerRelocation | capitalize })}}
       {{pagesMacros.tableRow({label: 'Relocation compliancy',             'data-qa': 'compliancy',              dataValue: data.relocationAndInjuries.prisonerCompliancy | capitalize })}}
-      {{pagesMacros.tableRow({label: 'HealthCare staff present',          'data-qa': 'healthcareStaffPresent',  dataValue: data.relocationAndInjuries.halthcareStaffPresent | title })}}
+      {{pagesMacros.tableRow({label: 'HealthCare staff present',          'data-qa': 'healthcareStaffPresent',  dataValue: data.relocationAndInjuries.healthcareStaffPresent | title })}}
```
Comment (petergphillips) [inline server/views/pages/check-answers.html]: Should either be `Healthcare` or `Health care`

**Pull request merged**